great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -78,19 +88,26 @@ def __init__(self, *args, **kwargs):
             "discard_subset_failing_expectations", False
         )
         boto3_options: dict = kwargs.pop("boto3_options", {})
+        azure_options: dict = kwargs.pop("azure_options", {})
 
-        # Try initializing boto3 client. If unsuccessful, we'll catch it when/if a S3BatchSpec is passed in.
+        # Try initializing cloud provider client. If unsuccessful, we'll catch it when/if a BatchSpec is passed in.
         try:
             self._s3 = boto3.client("s3", **boto3_options)
         except (TypeError, AttributeError):
             self._s3 = None
 
+        try:
+            self._azure = BlobServiceClient(**azure_options)
+        except (TypeError, AttributeError):
+            self._azure = None
+
```

Comment:
I don't love using `try/except` to evaluate each cloud provider. Probably not a big deal if we just have S3, Azure, and GCS but what are the implications this design has on future integrations with cloud providers?